### ISSUE TITLE
Multipass offers to bridge with networks that are not supported on qemu/macOS

### ISSUE THREAD

**Issue by lethargosapatheia** [canonical/multipass#2815]
**Describe the bug**
When I launch a VM, it's only after the creating that multipass checks if a network interface exists.

**To Reproduce**
```
multipass launch --cpus 4 --mem 8g --mount /Users/username/arbeit:/root/arbeit --network name=bridge0,mode=manual -n k8s-1 22.04
```
> launch failed: The following errors occurred:
qemu-system-aarch64: -nic vmnet-bridged,ifname=bridge0,model=virtio-net-pci,mac=52:54:00:4a:4f:ee: unsupported ifname 'bridge0', expected one of [ en0 ]

**Expected behavior**
To error out before creating anything

**Additional info**
 - OS: Ventura 13.0
- `multipass version` 1.10.1+mac
- `multipass info --all`
```
Name:           k8s-1
State:          Stopped
IPv4:           --
Release:        --
Image hash:     c363ebe92173 (Ubuntu 22.04 LTS)
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --
```
- `multipass get local.driver`
qemu

The same goes for other options like `mount`. If the folder doesn't exist, it will output the error, but the vm is still created. It seems intentional, actually, which to my mind is a bad idea, because you get something different than what you expect.

**Comment by ricab** [comment]
Hmm, I am unable to reproduce this. What does `multipass networks` show? 

This is what I see:

```
$ multipass networks
Name Type      Description
en0  ethernet  Ethernet
$ multipass launch --network en1
launch failed: Invalid arguments supplied
Invalid network options supplied
$ multipass launch --network name=en1,mode=manual
launch failed: Invalid arguments supplied
Invalid network options supplied
$ ifconfig | grep '^[[:alpha:]]'
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
en2: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
bridge100: flags=8a63<UP,BROADCAST,SMART,RUNNING,ALLMULTI,SIMPLEX,MULTICAST> mtu 1500
[...]
$ multipass launch --network name=bridge100,mode=manual
launch failed: Invalid arguments supplied
Invalid network options supplied
$ multipass launch --network name=bridge0,mode=manual
launch failed: Invalid arguments supplied
Invalid network options supplied
```

**Comment by ricab** [comment]
> So while this is actually behaving differently than how I initially thought it would, I would say that it still should probably check for compatibility, right?

Well, ideally, yes. In practice, there is only so much Multipass can reasonably check beforehand. At some point we have to give the network out to `qemu` and deal with errors if they arise.

> As far as the mount command is concerned, should I create another issue?

We did so already. It is linked just above your last comment. Thanks for reporting!

**Comment by ricab** [comment]
OK, I had a closer look into this and I agree it is a bug. The problem is that we offer a bunch of things in `multipass networks` that the vmnet infrastructure in qemu does not support. [Only physical interfaces](https://gitlab.com/qemu-project/qemu/-/blob/master/qemu-options.hx#L2826-L2829) are supported, so we should not be offering anything else. Once that is fixed, the command will reject other things early as it's supposed to.

I am surprised that your ethernet adapters are not included in the list of expected values in your error message (i.e. `[...] expected one of [ en0 ]`). Are those not physical?

**Comment by ricab** [comment]
It looks like [QEMU just asks vmnet](https://gitlab.com/qemu-project/qemu/-/blob/master/net/vmnet-bridged.m#L42) for allowed values. It woulld be great if Apple told us [what interfaces they support](https://developer.apple.com/documentation/vmnet/3152677-vmnet_copy_shared_interface_list)...

**Comment by lethargosapatheia** [comment]
Oh, after reading your initial response I tended to agree it wasn't a bug :)) But you obviously know what's going on. Thanks for taking a look!

I don't nave any physical ethernet adapters, if that's what you're asking. Now that I think about it, I'm not sure where they come form. I also have multipass installed among other things, but I  don't think that has any effect.
